### ISSUE TITLE
/usr/share/imapfilter/regex.lua:10: bad argument #1 to 'compile' (string expected, got nil)

### ISSUE THREAD

**Issue by sforshee** [lefcha/imapfilter#218]
I started seeing this when the imapfilter package for Ubuntu 21.04 was upgraded from 1:2.6.16-1 to 1:2.7.2-1. I see the following stacktrace:

```
$ /usr/bin/imapfilter -c ~/.imapfilter/mail.lua
imapfilter: /usr/share/imapfilter/regex.lua:10: bad argument #1 to 'compile' (string expected, got nil)
stack traceback:
        [C]: in ?
        [C]: in function 'compile'
        /usr/share/imapfilter/regex.lua:10: in function '__index'
        /usr/share/imapfilter/regex.lua:21: in function 'regex_search'
        /usr/share/imapfilter/mailbox.lua:986: in function 'match_field'
        /usr/share/imapfilter/set.lua:629: in function 'match_field'
        /home/sforshee/.imapfilter/mail.lua:15: in main chunk
```

**Comment by lefcha** [comment]
This was fixed in [v2.7.3](https://github.com/lefcha/imapfilter/releases/tag/v2.7.3), hopefully it will be upgraded sooner than later.

**Comment by sforshee** [comment]
I tested v2.7.3, unfortunately it does not seem to be fixed.

```
$ /usr/bin/imapfilter -V
IMAPFilter 2.7.3  Copyright (c) 2001-2020 Eleftherios Chatzimparmpas
$ /usr/bin/imapfilter -c ~/.imapfilter/mail.lua 
imapfilter: /usr/share/imapfilter/regex.lua:10: bad argument #1 to 'compile' (string expected, got nil)
stack traceback:
        [C]: in ?
        [C]: in function 'compile'
        /usr/share/imapfilter/regex.lua:10: in function '__index'
        /usr/share/imapfilter/regex.lua:21: in function 'regex_search'
        /usr/share/imapfilter/mailbox.lua:986: in function 'match_field'
        /usr/share/imapfilter/set.lua:629: in function 'match_field'
        /home/sforshee/.imapfilter/mail.lua:15: in main chunk
```

**Comment by lefcha** [comment]
Yes, you're right, this one looks different.  Can you try with the latest version, as it includes a fix: https://github.com/lefcha/imapfilter/commit/987ea442fa0aa3dc5f5d59ec89c9a3d818cd48d3

**Comment by lefcha** [comment]
Seems to fix it for me, so just released [v2.7.4](https://github.com/lefcha/imapfilter/releases/tag/v2.7.4). Thanks!

**Comment by sforshee** [comment]
It's fixed for me too. Thanks!
